addons/website_slides/controllers/main.py

```diff
@@ -508,6 +508,10 @@ def slide_channel_join(self, channel_id):
             return {'error': 'join_done'}
         return success
 
+    @http.route(['/slides/channel/leave'], type='json', auth='public', website=True)
+    def slide_channel_leave(self, channel_id):
+        request.env['slide.channel'].browse(channel_id).action_remove_member()
```

Comment:
why don't you call directly 
`request.env['slide.channel'].browse(channel_id)._remove_membership(request.env.user.partner_id)` ?